cpp/test/sg/fil_test.cu

```diff
@@ -47,9 +50,13 @@ struct FilTestParams {
   int num_cols   = 50;
   float nan_prob = 0.05;
   // forest parameters
-  int depth       = 8;
-  int num_trees   = 50;
-  float leaf_prob = 0.05;
+  int depth                      = 8;
+  int num_trees                  = 50;
+  float leaf_prob                = 0.05;
+  float node_categorical_prob    = 0.0;
```

Comment:
Could you add a short documentation comment describing each of the newly introduced parameters?